Add bio.tools/EDAM section to tool_xml.rst (#76)
galaxy-iuc/standards: docs/best_practices/tool_xml.rst

increased if this is likely to bring the two version in sync in a reasonable
time.

> 

Reviewer: ```suggestion
```

-   Ensure that you add at least one EDAM ``Topic`` and one EDAM ``Operation``
    to the entry (see below).

Tool annotations (EDAM)
---------------------

The bio.tools registry makes use of the `EDAM`_ ontology
to annotate tools with ``Data``, ``Format``, ``Operation`` and ``Topic`` terms.

`EDAM`_ terms are used to give a description of the tool's scientific domain and the
functionalities it provides. The `EDAM browser <https://edamontology.github.io/edam-browser/>`__
makes the process of picking EDAM terms much easier.

Once a bio.tools identifier is added as an ``xref`` in the tool wrapper, the easiest way
to improve the available EDAM annotations is to update the bio.tools entry metadata.

Alternatively, you can add EDAM annotations directly to a Galaxy tool by adding the
`edam_topics <https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-edam-topics>`__
and `edam_operations <https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-edam-operations>`__
elements to the tool XML file. This approach is not recommended though, as it risks duplicating the
tool annotation work in multiple places.

> When picking EDAM terms, avoid `root terms` such as `Topic` and `Operation`, and pick the

Reviewer: ```suggestion
When picking EDAM terms, avoid "root terms" such as ``Topic`` and ``Operation``, and pick the
```

It is important to cross-link Galaxy tools to best practice metadata in registries and
catalogs such as `bio.tools <https://bio.tools/>`__.

bio.tools is a global portal for bioinformatics resources that helps researchers to find,
understand, compare and select resources suitable for their work.

You can link a Galaxy tool to its bio.tools entry by adding a ``xref`` snippet
to the tool wrapper: i.e. modifying the ``*.xml`` file. Below is an example ``xml`` for
`Racon <https://github.com/bgruening/galaxytools/tree/1570f3a28232b4b88385cdfbb68f79d80ff1dabb/tools/racon>`__

.. code:: xml
    <macros>
        <import>macros.xml</import>
    </macros>
    <xrefs>
        <xref type="bio.tools">Racon</xref>
    </xrefs>
    <expand macro="requirements" />
    <version_command>racon --version</version_command>
    <command detect_errors="exit_code"><![CDATA[

Please reference only the tool which is wrapped, not the dependencies.

If a bio.tools entry does not exist, you should add an entry as follows:

-   Create an account on bio.tools.
-   Add a new tool, completing as much metadata as possible / available.
-   Ensure that you add at least one EDAM ``Topic`` and one EDAM ``Operation``

> to the entry (see below).

Reviewer: ```suggestion
-   Ensure that you add at least one EDAM ``Topic`` and one EDAM ``Operation``
    to the entry (see below).
```

Tool annotations (EDAM)
---------------------

> The bio.tools registry makes use of the `EDAM ontology <http://edamontology.org/page>`__

Reviewer: ```suggestion
The bio.tools registry makes use of the `EDAM`_ ontology
```

to annotate tools with ``Data``, ``Format``, ``Operation`` and ``Topic`` terms.

`EDAM`_ terms are used to give a description of the tool's scientific domain and the

> functionalities it provides. `EDAM browser <https://edamontology.github.io/edam-browser/>`__

Reviewer: ```suggestion
functionalities it provides. The `EDAM browser <https://edamontology.github.io/edam-browser/>`__
```

makes the process of picking EDAM terms much easier.

Once a bio.tools identifier is added as an ``xref`` in the tool wrapper, the easiest way
to improve the available EDAM annotations is to update the bio.tools entry metadata.

> If you wish to add EDAM annotations directly to a Galaxy tool wrapper, the two options are:
> 

Reviewer: ```suggestion
Alternatively, you can add EDAM annotations directly to a Galaxy tool by adding the
`edam_topics <https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-edam-topics>`__
and `edam_operations <https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-edam-operations>`__
elements to the tool XML file. This approach is not recommended though, as it risks duplicating the
tool annotation work in multiple places.
```

> - identify the EDAM annotations available in existing `bio.tools`_ entries for your tools: these
> will include ``Topics`` and ``Operations`` considered relevant for the tool, or
> - if the tool is not in bio.tools, you will need to add a new tool and
> annotate the entry with the relevant EDAM annotations.

Reviewer: ```suggestion
```

When picking EDAM terms, avoid `root terms` such as `Topic` and `Operation`, and pick the
most specific terms available. If you feel like the terms required to describe the tool are

> missing, do not hesitate to [ask for new terms](https://edamontologydocs.readthedocs.io/en/latest/getting_involved.html#suggestions-requests).

Reviewer: ```suggestion
missing, do not hesitate to `ask for new terms <https://edamontologydocs.readthedocs.io/en/latest/getting_involved.html#suggestions-requests>`__.
```
